### PR TITLE
Add recipe for oslo.concurrency

### PR DESCRIPTION
**Pull request by sy-hbk**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by pg-hbk** [comment]
I'm fine with being a maintainer.

File: recipes/oslo.concurrency/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - pip
+    - python
+  run:
+    - pbr!=2.1.0,>=2.0.0
+    - oslo.config>=5.2.0
+    - oslo.i18n>=3.15.3
+    - oslo.utils>=3.33.0
+    - six>=1.10.0
+    - fasteners>=0.7.0
+    - enum34>=1.0.4  # [py2k]
```
Comment (scopatz) [inline recipes/oslo.concurrency/meta.yaml]: There needs to be a single space between the package name and the pin, e.g. `pbr !=2.1.0,>=2.0.0`

File: recipes/oslo.concurrency/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: d5f5ccca934a7e736b934044e1a347da2e139db6e4df3a2ac5dedc15e5287c81
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (scopatz) [inline recipes/oslo.concurrency/meta.yaml]: Can this be a `noarch: python` recipe?

Comment (sy-hbk) [reply recipes/oslo.concurrency/meta.yaml]: No this is not a `noarch` recipe.

File: recipes/oslo.concurrency/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{% set name = "oslo.concurrency" %}
```
Comment (scopatz) [inline recipes/oslo.concurrency/meta.yaml]: Hmm I didn't think "." was allowed in package names. CC @conda-forge/staged-recipes

Comment (sy-hbk) [reply recipes/oslo.concurrency/meta.yaml]: I think "." can be allowed here in package name. Example-
https://github.com/conda-forge/oslo.config-feedstock/blob/master/recipe/meta.yaml
https://github.com/conda-forge/oslo.utils-feedstock/blob/master/recipe/meta.yaml

Comment (scopatz) [reply recipes/oslo.concurrency/meta.yaml]: Is there a non-oslo package example?

Comment (isuruf) [reply recipes/oslo.concurrency/meta.yaml]: ruamel.yaml and backports packages

Comment (scopatz) [reply recipes/oslo.concurrency/meta.yaml]: OK

**Pull request merged**